### PR TITLE
[excel] (Worksheet) Page layout snippet

### PR DESCRIPTION
**Pull request by AlexJerabek**
This snippet showcases some page layout and print setting features coming in Excel 1.9.

### CODE REVIEW

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        </section>
+        <section class="setup ms-font-m">
+          <h3>Setup</h3>
+          <button id="setup" class="ms-Button">
+                <span class="ms-Button-label">Create sample data</span>
+            </button>
+        </section>
+        <section class="setup ms-font-m">
+            <h3>Try it out</h3>
+            <p>Go to the <b>View > Page Layout</b> to see the layout changes as you press the buttons below.</p>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Consider rephrasing this instruction ->

`<p>In Excel, choose <b>View > Page Layout</b>, then observe the page layout changes as you press the following buttons:</p>`

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            <button id="setPageBreaks" class="ms-Button">
+                <span class="ms-Button-label">Set page break</span>
+            </button><p/>
+            <button id="setPrintTitleRow" class="ms-Button">
+                <span class="ms-Button-label">Set title rows on each page</span>
+            </button><p/><p/>
+            <button id="center" class="ms-Button">
+                <span class="ms-Button-label">Center data</span>
+            </button>
+            <p>Go to the <b>File > Print</b> to see differences in printing before and after selecting the options below.</p>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Consider rephrasing this instruction -> 

`<p>In Excel, choose <b>File > Print</b> to see differences in printing before and after you press the following buttons:</p>`

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+          <h3>Setup</h3>
+          <button id="setup" class="ms-Button">
+                <span class="ms-Button-label">Create sample data</span>
+            </button>
+        </section>
+        <section class="setup ms-font-m">
+            <h3>Try it out</h3>
+            <p>Go to the <b>View > Page Layout</b> to see the layout changes as you press the buttons below.</p>
+            <button id="setPageBreaks" class="ms-Button">
+                <span class="ms-Button-label">Set page break</span>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Suggest changing button label -> Add page break

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            </button><p/>
+            <button id="setPrintTitleRow" class="ms-Button">
+                <span class="ms-Button-label">Set title rows on each page</span>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Suggest changing button label -> Add title row to each page  (or maybe this instead? -> Add title row to all pages)

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            </button><p/><p/>
+            <button id="center" class="ms-Button">
+                <span class="ms-Button-label">Center data</span>
+            </button>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: I'm not seeing any noticeable changes in "Page Layout" view when the "Center data" button is clicked. However, if I choose File > Print, I can see that the data has been centered (horizontally and vertically). Therefore, it seems like this button belongs in the next section of the task pane (since you only see the alignment changes in File > Print, and not in View > Page Layout).

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 6]
```diff
@@ -0,0 +1,198 @@
+order: 5
+id: excel-worksheet-page-layout
+name: Page layout
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Seems like "Page layout" is only part of the story for this snippet...some of the functions control page layout, others control print settings. To improve discoverability for this snippet, would it be too wordy to name it "Page layout and print settings" ?   Or maybe just "Print settings" is sufficient?

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+description: Sets the page layout for printing a worksheet.
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: If you change the snippet name, consider also changing the description accordingly, e.g. -> "Changes page layout and other settings for printing a worksheet."

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+            } catch (error) {
+                OfficeHelpers.UI.notify(error);
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+          <p>This sample shows how to set page layouts for printing a worksheet.</p>
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: If you change the snippet name/description, consider also changing this HTML description accordingly, e.g. -> "This sample shows how to change page layout and other settings for printing a worksheet."

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml (hunk 5 above)
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/worksheet-page-layout.yaml]: The centering doesn't show up until the screen refreshes. I flipped some of the options so a user will at least see the changes in View > Page Layout before switching away.

**Comment by AlexJerabek** [comment]
Thanks for the review @kbrandl. I updated the description according to your suggestion and made the other little changes.

File: samples/excel/85-preview-apis/worksheet-page-layout.yaml [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+                <span class="ms-Button-label">Create sample data</span>
+            </button>
+        </section>
+        <section class="setup ms-font-m">
+            <h3>Try it out</h3>
+            <p>In Excel, choose <b>View > Page Layout</b>, then observe the page layout changes as you press the following buttons.</p>
+            <button id="setPageBreaks" class="ms-Button">
+                <span class="ms-Button-label">Add page break</span>
+            </button><p/>
+            <button id="center" class="ms-Button">
```
Comment (kbrandl) [inline samples/excel/85-preview-apis/worksheet-page-layout.yaml]: Still seeing weird behavior when I click this button in the sequence it's presented. In **View > Page Layout** mode, when I click the "Center data" button immediately after clicking the "Add page break" button, the first page of data appears to be centered vertically (eventually, after the screen refreshes) but the second page does not appear to be centered vertically. However, if I choose **File > Print** I can see in the print preview that both pages of data are centered. Seems like this flaky behavior could be avoided if you move this button down to the next section of the task pane (before the "Set to black and white printing" button)?
